lib/AnimationStation/src/Effects/CustomTheme.cpp

```diff
@@ -1,41 +1,108 @@
+            hitColor.insert_or_assign(matrix.pixels[r][c].index, defaultColor);            
+        }
+    }
+}
 
+void CustomTheme::UpdatePixels(std::vector<Pixel> inpixels) {
+  this->pixels = inpixels;
 }
 
 void CustomTheme::Animate(RGB (&frame)[100]) {
-  for (size_t r = 0; r != matrix->pixels.size(); r++) {
-    for (size_t c = 0; c != matrix->pixels[r].size(); c++) {
-      if (matrix->pixels[r][c].index == NO_PIXEL.index)
-        continue;
-
-      auto itr = theme.find(matrix->pixels[r][c].mask);
-      if (itr != theme.end())
-        for (size_t p = 0; p != matrix->pixels[r][c].positions.size(); p++)
-          frame[matrix->pixels[r][c].positions[p]] = itr->second;
-      else
-        for (size_t p = 0; p != matrix->pixels[r][c].positions.size(); p++)
-          frame[matrix->pixels[r][c].positions[p]] = defaultColor;
+
+    coolDownTimeInMs = AnimationStation::options.customThemeCooldownTimeInMs;
+
+    for (size_t p = 0; p < pixels.size(); p++) {
+        if (pixels[p].index != NO_PIXEL.index) {
+            times[pixels[p].index] = coolDownTimeInMs;               
+            hitColor[pixels[p].index] = frame[pixels[p].positions[0]];       
+        }
+    }
+
+    updateTimeInms = absolute_time_diff_us(lastUpdateTime, get_absolute_time()) / 1000;
+
+    for (size_t r = 0; r != matrix->pixels.size(); r++) {
+        for (size_t c = 0; c != matrix->pixels[r].size(); c++) {
+            if (matrix->pixels[r][c].index == NO_PIXEL.index)
+                continue;
+
+            // Count down the timer
+            times[matrix->pixels[r][c].index] -= updateTimeInms;
+            if (times[matrix->pixels[r][c].index] < 0) {
+                times[matrix->pixels[r][c].index] = 0;
+            };
+
+            auto itr = theme.find(matrix->pixels[r][c].mask);
+            if (itr != theme.end()) {
+                for (size_t p = 0; p != matrix->pixels[r][c].positions.size(); p++) {
+                    // Interpolate from hitColor (color the button was assigned when pressed) back to the theme color
+                    frame[matrix->pixels[r][c].positions[p]] = BlendColor(hitColor[matrix->pixels[r][c].index], itr->second, times[matrix->pixels[r][c].index]);
+                }
+            } else {
+                for (size_t p = 0; p != matrix->pixels[r][c].positions.size(); p++) {
+                    frame[matrix->pixels[r][c].positions[p]] = defaultColor;
+                }
+            }
+        }
     }
-  }
+
+    lastUpdateTime = get_absolute_time();
 }
 
 bool CustomTheme::HasTheme() {
-  return CustomTheme::theme.size() > 0;
+    return CustomTheme::theme.size() > 0;
 }
 
 void CustomTheme::SetCustomTheme(std::map<uint32_t, RGB> customTheme) {
-  CustomTheme::theme = customTheme;
-  AnimationStation::effectCount = TOTAL_EFFECTS + 1;
+    CustomTheme::theme = customTheme;
+    AnimationStation::effectCount = TOTAL_EFFECTS + 1;
 }
 
+#define PRESS_COOLDOWN_INCREMENT   500
```

Comment:
stick these at the top of the C++ file as well, makes it much easier for someone else to come in and find them